ocs/ocs_agent.py

```diff
@@ -237,30 +240,62 @@ def onLeave(self, details):
         self.subscribed_topics = set()
         self.realm_joined = False
 
+    def _shutdown(self):
+        # Stop all sessions and then stop the reactor
+        self._stop_all_running_sessions()
+        try:
+            self.log.info('stopping reactor')
+            reactor.stop()
+        except ReactorNotRunning:
+            pass
+
     @inlineCallbacks
     def onDisconnect(self):
         self.log.info('transport disconnected')
+        self.log.info('waiting for reconnection')
 
         # Wait to see if we reconnect before stopping the reactor
-        timeout = 10
+        timeout = self.site_args.crossbar_timeout
+
+        # Define signal handlers to interrupt during wait for reconnection
+        def signal_handler(sig, frame):
+            self.log.info('caught {signal}!', signal=signal.Signals(sig).name)
+            self._shutdown()
+
+        signal.signal(signal.SIGINT, signal_handler)
+        signal.signal(signal.SIGTERM, signal_handler)
+
+        # Wait forever
+        if timeout == 0:
+            while True:
+                if self.realm_joined:
+                    return
+
+                yield dsleep(1)
```

Comment:
I think this is equivalent to:
```
if timeout == 0:
    return
```

Reply:
Agreed, simplified in https://github.com/simonsobs/ocs/pull/337/commits/514fd3497835c98bc4c2188017cab6444b3c2463